pyuvdata/ms.py

```diff
@@ -0,0 +1,235 @@
+        history_str='APP_PARAMS;CLI_COMMAND;APPLICATION;MESSAGE;OBJECT_ID;OBSERVATION_ID;ORIGIN;PRIORITY;TIME\n'
+        #string to store special casa history
+        app_params=history_table.getcol('APP_PARAMS')['array']
+        cli_command=history_table.getcol('CLI_COMMAND')['array']
+        application=history_table.getcol('APPLICATION')
+        message=history_table.getcol('MESSAGE')
+        obj_id=history_table.getcol('OBJECT_ID')
+        obs_id=history_table.getcol('OBSERVATION_ID')
+        origin=history_table.getcol('ORIGIN')
+        priority=history_table.getcol('PRIORITY')
+        times=history_table.getcol('TIME')
+        #Now loop through columns and generate history string
+        ntimes=len(times)
+        if test_import_uvfits:
+            ntimes-=5
+        for tbrow in range(ntimes):
+            message_str+=str(message[tbrow])
+            newline=str(app_params[tbrow]) \
+            +';'+str(cli_command[tbrow]) \
+            +';'+str(application[tbrow]) \
+            +';'+str(message[tbrow]) \
+            +';'+str(obj_id[tbrow]) \
+            +';'+str(obs_id[tbrow]) \
+            +';'+str(origin[tbrow]) \
+            +';'+str(priority[tbrow]) \
+            +';'+str(times[tbrow])+'\n'
+            history_str+=newline
+            if tbrow<ntimes-1:
+                message_str+='\n'
+        return message_str,history_str
+
+
+    
+    #ms write functionality to be added later. 
+    def write_ms(self):
+        '''
+        writing ms is not yet supported
+        '''
+    
+    def read_ms(self,filepath,run_check=True,run_check_acceptability=True,data_column='DATA',pol_order='AIPS',test_import_uvfits=False):
+        '''
+        read in a casa measurement set
+        ARGS:
+        filepath: name of the measurement set folder
+        run_check:specify whether you want to run check
+        run_check_acceptability: run sanity check for new UVData object
+        data_column: specify which CASA measurement set data column to read from (can be 'DATA','CORRECTED', or 'MODEL')
+        pol_order: use 'AIPS' or 'CASA' ordering of polarizations?
+        test_import_uvfits: test a .ms file that was created with CASA importuvfits method: 
+        ignores last five lines of history which contain information on importuvfits command. 
+        '''
+        #make sure user requests a valid data_column
+        if data_column!='DATA' and data_column!='CORRECTED_DATA' and data_column!='MODEL':
+            raise ValueError('Invalid data_column value supplied. Use \'Data\',\'MODEL\' or \'CORRECTED_DATA\'')
+        if not os.path.exists(filepath):
+            raise(IOError, filepath + ' not found')
+        #set visibility units
+        if(data_column=='DATA'):
+            self.vis_units="UNCALIB"
+        elif(data_column=='CORRECTED_DATA'):
+            self.vis_units="JY"
+        elif(data_column=='MODEL'):
+            self.vis_units="JY"
+        self.data_column=data_column
```

Comment:
I understand why you need this as an argument to the read_ms method, but why do you want to attach it to the object? Does a user need this information to stick around for some reason? If so, I'd suggest adding it to the extra_keywords dictionary rather than as a new attribute on the object (and on the UVData object).